--- a/src/libgm2/libm2cor/Makefile.am
+++ b/src/libgm2/libm2cor/Makefile.am
@@ -27,7 +27,7 @@ MAKEOVERRIDES=
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(target_alias)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- a/src/libgm2/libm2cor/Makefile.in
+++ b/src/libgm2/libm2cor/Makefile.in
@@ -369,7 +369,7 @@ MAKEOVERRIDES =
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(target_alias)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test x$(MULTIOSDIR) != x.; then echo /$(MULTIOSDIR); fi)
--- a/src/libgm2/libm2iso/Makefile.am
+++ b/src/libgm2/libm2iso/Makefile.am
@@ -27,7 +27,7 @@ MAKEOVERRIDES=
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(target_alias)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- a/src/libgm2/libm2iso/Makefile.in
+++ b/src/libgm2/libm2iso/Makefile.in
@@ -386,7 +386,7 @@ MAKEOVERRIDES =
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(target_alias)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test x$(MULTIOSDIR) != x.; then echo /$(MULTIOSDIR); fi)
--- a/src/libgm2/libm2log/Makefile.am
+++ b/src/libgm2/libm2log/Makefile.am
@@ -27,7 +27,7 @@ MAKEOVERRIDES=
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(target_alias)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- a/src/libgm2/libm2log/Makefile.in
+++ b/src/libgm2/libm2log/Makefile.in
@@ -374,7 +374,7 @@ MAKEOVERRIDES =
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(target_alias)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test x$(MULTIOSDIR) != x.; then echo /$(MULTIOSDIR); fi)
--- a/src/libgm2/libm2min/Makefile.am
+++ b/src/libgm2/libm2min/Makefile.am
@@ -27,7 +27,7 @@ MAKEOVERRIDES=
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(target_alias)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- a/src/libgm2/libm2min/Makefile.in
+++ b/src/libgm2/libm2min/Makefile.in
@@ -372,7 +372,7 @@ MAKEOVERRIDES =
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(target_alias)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test x$(MULTIOSDIR) != x.; then echo /$(MULTIOSDIR); fi)
--- a/src/libgm2/libm2pim/Makefile.am
+++ b/src/libgm2/libm2pim/Makefile.am
@@ -27,7 +27,7 @@ MAKEOVERRIDES=
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(target_alias)/$(version)
 # Used to install the shared libgcc.
 slibdir = @slibdir@
 
--- a/src/libgm2/libm2pim/Makefile.in
+++ b/src/libgm2/libm2pim/Makefile.in
@@ -382,7 +382,7 @@ MAKEOVERRIDES =
 version := $(shell $(CC) -dumpversion)
 
 # Directory in which the compiler finds libraries etc.
-libsubdir = $(libdir)/gcc/$(target_alias)/$(version)
+libsubdir = $(libdir)/gcc-cross/$(target_alias)/$(version)
 MULTIDIR := $(shell $(CC) $(CFLAGS) -print-multi-directory)
 MULTIOSDIR := $(shell $(CC) $(CFLAGS) -print-multi-os-directory)
 MULTIOSSUBDIR := $(shell if test x$(MULTIOSDIR) != x.; then echo /$(MULTIOSDIR); fi)
